avogadro/core/molecule.cpp

```diff
@@ -258,6 +258,7 @@ MatrixX Molecule::partialCharges(const std::string& type) const
     return search->second; // value from the map
   } else {
     MatrixX charges(atomCount(), 1);
+    charges.fill(0.0);
```

Comment:
I *think* you should be able to do:
```
MatrixX charges(atomCount(), 1, 0.0);
```

Otherwise, this is fine.

Reply:
Tried it now. Seems to only work for sized vectors.